ballerina/modules/parser/state.bal

```diff
@@ -62,9 +62,12 @@ public class ParserState {
 
     common:Event[] eventBuffer = [];
 
-    public isolated function init(string[] lines) returns ParsingError? {
-        self.lines = lines;
-        self.numLines = lines.length();
+    boolean isString;
```

Comment:
Shall we rename this flag to be more specific on its purpose (having just `isString` looks a bit ambiguous to me)? WDYT?
```suggestion
    boolean isStringInput;
```